### PR TITLE
Make Rust version no_std and improve docs

### PR DESCRIPTION
**Pull request by Keavon**

### CODE REVIEW

File: Rust/src/lib.rs [hunk 1]
```diff
@@ -493,6 +497,24 @@ impl FastNoiseLite {
...
+        }
+    }
+
+    /// Accurate to within roughly 6% at worst.
+    ///
+    /// Note: Requires f >= 0.
+    #[inline(always)]
+    fn fast_sqrt(f: f32) -> f32 {
+        // Explanation: https://stackoverflow.com/questions/41785416/how-does-this-sqrt-approximation-inline-assembly-function-work
+        f32::from_bits((f.to_bits() + 0x3f80_0000) >> 1)
```
Comment (Keavon) [inline Rust/src/lib.rs:515]: Alternatively, we could port [this](https://github.com/simonbyrne/apple-libm/blob/master/Source/ARM/sqrt.c) for the accurate implementation?

We should maybe also consider switching between the std version and the no_std polyfill? But then the results would differ, so I'm not sure what to do exactly.

Comment (Keavon) [reply Rust/src/lib.rs:515]: Another option is to use this to refine the guess (it doesn't seem to iterate for more than several iterations, but it'll go until the numerical instability reaches insignificance and the guesses stop changing):

```rs
let mut guess = f32::from_bits((f.to_bits() + 0x3f80_0000) >> 1);

loop {
    let new_guess = 0.5 * (guess + f / guess);
    if new_guess == guess {
        break;
    }
    guess = new_guess;
}

guess
```

File: Rust/src/lib.rs [hunk 2]
```diff
@@ -493,6 +497,47 @@ impl FastNoiseLite {
...
+            loop {
+                let new_guess = 0.5 * (guess + f / guess);
+                if new_guess == guess {
+                    break;
+                }
+                guess = new_guess;
+            }
+
+            guess
+        };
```
Comment (Keavon) [inline Rust/src/lib.rs:533]: Alternatively, we could port [this](https://github.com/simonbyrne/apple-libm/blob/master/Source/ARM/sqrt.c) for the accurate implementation?

**Comment by Auburn** [comment]
Could you pr these changes into the v1.1 branch, I'm getting ready to merge that to master

**Comment by Keavon** [comment]
I updated the branch and version numbering. However upon changing the base to merge this into, it looks like it created conflicts. I'll have to wait a few hours or possibly until tomorrow until I can resolve those.

**Comment by Keavon** [comment]
I updated the branch and version numbering. However upon changing the base to merge this into, it looks like it created conflicts. I'll have to wait a few hours or possibly until tomorrow until I can resolve those.

**Pull request merged**